src/test/java/io/vlingo/xoom/schemata/queries/CountingStateStore.java

```diff
@@ -0,0 +1,47 @@
+package io.vlingo.xoom.schemata.queries;
+
+import io.vlingo.xoom.common.Completes;
+import io.vlingo.xoom.reactivestreams.Stream;
+import io.vlingo.xoom.symbio.store.QueryExpression;
+import io.vlingo.xoom.symbio.store.state.inmemory.InMemoryStateStoreActor;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class CountingStateStore extends InMemoryStateStoreActor {
```

Comment:
Loose suggestion: 

Instead of extending the `InMemoryStateStoreActor` we could make the `CountingStateStore` a decorator of `StateStore`. 

This way it would be possible to use `CountingStateStore` with any implementation of `StateStore`. Implementation would be very similar, but isntead of calling methods on `super` we'd be calling methods on the decorated `stateStore`.

It would be instantiated like this: `new CountingStateStore(new InMemoryStateStore(Arrays.asList(new NoOpDispatcher())))`.

Reply:
@jakzal Good idea.

Reply:
Fixed by ca39cb7